samples/bluetooth/mesh/chat/src/model_handler.c

```diff
@@ -0,0 +1,424 @@
+
+			err = bt_mesh_chat_cli_presence_status_pub(&chat, NULL,
+								   &presence);
+			if (err) {
+				LOG_WRN("Failed to update presence: %d", err);
+			}
+
+			/* Print own presence in the chat. */
+			uart_handler_tx("---> you are now %s",
+					presence_string[presence.presence]);
+
+			return;
+		}
+	}
+
+	uart_handler_tx(
+		"Unknown presence status: %s. Possible presence statuses:",
+		data);
+	for (i = 0; i < ARRAY_SIZE(presence_string); i++) {
+		uart_handler_tx("%s", presence_string[i]);
+	}
+}
+
+/* .. include_startingpoint_model_handler_rst_1 */
+static void uart_cmd_direct_message(const uint8_t *data, uint8_t len)
+{
+	struct bt_mesh_msg_ctx ctx;
+	struct bt_mesh_model_pub *pub = chat.model->pub;
+	struct bt_mesh_chat_cli_message msg;
+	uint16_t addr;
+	int err;
+
+	if (len < 6) {
+		uart_handler_tx("Incorrect command length. Type /help to see how to use this command.");
+		return;
+	}
+
+	addr = strtol(&data[2], NULL, 16);
+
+	msg.msg = &data[7];
+
+	/* Fill message context parameters manually. */
+	ctx.addr = addr;
+	ctx.send_ttl = pub->ttl;
+	ctx.send_rel = pub->send_rel;
+	ctx.app_idx = 0;
+
+	/* Print own message to the chat. */
+	uart_handler_tx("<you>: *0x%04X* %s", addr, &data[7]);
+
+	err = bt_mesh_chat_cli_message_send(&chat, &ctx, &msg, true);
+	if (err) {
+		LOG_WRN("Failed to publish message: %d", err);
+	} else {
+		uart_reply_print(addr);
+	}
+}
+/* .. include_endpoint_model_handler_rst_1 */
+
+static void uart_cmd_presence_get(const uint8_t *data, uint8_t len)
+{
+	struct bt_mesh_msg_ctx ctx = { 0 };
+	struct bt_mesh_model_pub *pub = chat.model->pub;
+	struct bt_mesh_chat_cli_presence_status presence;
+	uint16_t addr;
+	int err;
+
+	if (len < 6) {
+		uart_handler_tx("Incorrect command length. Type /help to see how to use this command.");
+		return;
+	}
+
+	addr = strtol(&data[2], NULL, 16);
+
+	/* Fill message context parameters manually. */
+	ctx.addr = addr;
+	ctx.send_ttl = pub->ttl;
+	ctx.send_rel = pub->send_rel;
+	ctx.app_idx = 0;
+
+	err = bt_mesh_chat_cli_presence_get(&chat, &ctx, &presence);
+	if (err) {
+		LOG_WRN("Failed to publish message: %d", err);
+	} else {
+		uart_handler_tx("---> 0x%04X is %s", addr,
+				presence_string[presence.presence]);
+	}
+}
+
+static void uart_cmd_help(const uint8_t *data, uint8_t len);
+
+struct uart_cmd_handler {
+	const char *name;
+	const char *args;
+	const char *help;
+	void (*handler)(const uint8_t *data, uint8_t len);
+};
+
+static const struct uart_cmd_handler uart_cmd_handlers[] = {
+	{"/help", "", "Prints help message", uart_cmd_help},
+	{"/presence set ", "{available,away,dnd,inactive}",
+			"Sets presence status of the current node",
+			uart_cmd_presence_set},
+	{"/presence get ", "{node}",
+			"Obtains presence status of the remote node",
+			uart_cmd_presence_get},
+	{"/msg ", "{node} {msg}", "Sends direct message to a remote node.",
+			uart_cmd_direct_message},
+	{"/chat ", "{message}", "Sends message to a chat.", uart_cmd_send},
+};
```

Comment:
Could also look into using the Zephyr shell module for this, it's quite similar. Not a blocker though.